src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticBraceCompletionTests.cs

```diff
@@ -1665,6 +1661,35 @@ public void man()
             CheckReturn(session.Session, 12, expected);
         }
 
+        [WpfFact, WorkItem("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1758005")]
+        public void NoFormattingAfterNewlineIfOptionsDisabled()
+        {
+            var code = @"namespace NS1
+$$";
+
+            var expected = @"namespace NS1
+{}";
+
+            var expectedAfterReturn = @"namespace NS1
+{
+
+}";
+
+            var globalOptions = new OptionsCollection(LanguageNames.CSharp)
+            {
+                { FormattingOptions2.SmartIndent, FormattingOptions2.IndentStyle.None },
+                { AutoFormattingOptionsStorage.FormatOnCloseBrace, false },
+            };
+
+            using var session = CreateSession(code, globalOptions);
+            Assert.NotNull(session);
+
+            CheckStart(session.Session);
+            Assert.Equal(expected, session.Session.SubjectBuffer.CurrentSnapshot.GetText());
+
+            CheckReturn(session.Session, 0, expectedAfterReturn);
```

Comment:
can you clarify what is was about this prior code that was causing the problem?

Reply:
We previous assume calling formatter on the brace pair (including added newline in-between) would always cause text changes. But it's not true with certain option combination, which would cause us the throw.

Reply:
thanks!